AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java

```diff
@@ -3493,49 +3493,89 @@ private void showDeveloperContact() {
+                UIUtils.showThemedToast(AbstractFlashcardViewer.this, getString(R.string.valid_js_api_version, sCurrentJsApiVersion), false);
+                throw new IllegalArgumentException("Invalid version format");
+            }
+            this.version = version;
+        }
 
-        if (versionSupplied.length < 3 || versionSupplied.length > 4) {
-            UIUtils.showThemedToast(AbstractFlashcardViewer.this, getString(R.string.valid_js_api_version, sCurrentJsApiVersion), false);
-            return false;
-        } else {
-            JsApiVersionClass mJsSuppliedVersion = new JsApiVersionClass();
+        @Override
+        public int compareTo(Version that) {
+            if (that == null) {
+                return 1;
+            }
 
-            mJsSuppliedVersion.major = Integer.parseInt(versionSupplied[0]);
-            mJsSuppliedVersion.minor = Integer.parseInt(versionSupplied[1]);
-            mJsSuppliedVersion.patch = Integer.parseInt(versionSupplied[2]);
+            String[] thisParts = this.get().split("\\.");
+            String[] thatParts = that.get().split("\\.");
+            int length = Math.max(thisParts.length, thatParts.length);
 
-            if (mJsCurrentVersion.major > mJsSuppliedVersion.major || mJsCurrentVersion.minor > mJsSuppliedVersion.minor || mJsCurrentVersion.patch > mJsSuppliedVersion.patch) {
-                UIUtils.showThemedToast(AbstractFlashcardViewer.this, getString(R.string.update_js_api_version, sCurrentJsApiVersion), false);
-                return false;
-            } else if (mJsCurrentVersion.major == mJsSuppliedVersion.major && mJsCurrentVersion.minor == mJsSuppliedVersion.minor && mJsCurrentVersion.patch == mJsSuppliedVersion.patch) {
+            for (int i = 0; i < length; i++) {
+                int thisPart = i < thisParts.length ? Integer.parseInt(thisParts[i]) : 0;
+                int thatPart = i < thatParts.length ? Integer.parseInt(thatParts[i]) : 0;
+
+                if (thisPart < thatPart) {
+                    return -1;
+                }
+
+                if (thisPart > thatPart) {
+                    return 1;
+                }
+            }
+            return 0;
+        }
+
+        @Override
+        public boolean equals(Object that) {
+            if (this == that) {
                 return true;
-            } else {
-                UIUtils.showThemedToast(AbstractFlashcardViewer.this, getString(R.string.valid_js_api_version, sCurrentJsApiVersion), false);
+            }
+
+            if (that == null) {
                 return false;
             }
+
+            if (this.getClass() != that.getClass()) {
+                return false;
+            }
+
+            return this.compareTo((Version) that) == 0;
+        }
+    }
+
+    private boolean requireApiVersion(String apiVer) {
+
+        Version mVersionCurrent = new Version(sCurrentJsApiVersion);
+        Version mVersionSupplied = new Version(apiVer);
```

Comment:
What if this throws?

Reply:
Not a member variable - no need for the m prefix
```suggestion
        Version versionSupplied = new Version(apiVer);
```